org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/client/IEditor.java

```diff
@@ -15,10 +14,38 @@
      * Get the input from the user, optionally spanning multiple lines.
      *
      * @return The input typed in by the user.
-     * @throws IOException
-     *             When an IO error occurs.
      */
-    String getInput() throws IOException;
+    String getInput();
+
+    /**
+     * Set the prompt to display.
+     *
+     * @param promptString
+     *            The prompt string.
+     */
+    void setPrompt(StyledText promptString);
+
+    /**
+     * Returns the prompt used by this EclipseEditor.
+     *
+     * @return The prompt used by this EclipseEditor.
+     */
+    StyledText getPrompt();
```

Comment:
I added this getter and the one below such that the `EclipseRepl` class has access to the prompts when it needs to append the input to the `EclipseDisplay`.